src/extensions/extensionsHelper.ts

```diff
@@ -596,16 +785,21 @@ export async function uploadExtensionVersionFromGitHubSource(args: {
   force: boolean;
 }): Promise<ExtensionVersion | undefined> {
   const extensionRef = `${args.publisherId}/${args.extensionId}`;
-
-  // Check if Extension already exists.
   let extension: Extension | undefined;
+  let latestVersion: ExtensionVersion | undefined;
   try {
     extension = await getExtension(extensionRef);
+    latestVersion = await getExtensionVersion(`${extensionRef}@latest`);
   } catch (err: any) {
-    // Silently fail and continue the publish flow if Extension not found.
+    // Silently fail and continue if extension is new or has no latest version set.
   }
+  displayExtensionHeader(extensionRef, extension, latestVersion?.extensionRoot);
 
-  // Prompt for repo URI and validate that it hasn't changed if previously set.
+  if (args.stage && !stageOptions.includes(args.stage)) {
+    throw new FirebaseError(`--stage flag only supports the following values: ${stageOptions}`);
```

Comment:
Super nit - this currently prints out with no spaces between the entries in stage options
```suggestion
    throw new FirebaseError(`--stage flag only supports the following values: ${stageOptions.join(', ')}`);
```